make/common/Modules.gmk

```diff
@@ -310,7 +310,7 @@ MODULE_DEPS_MAKEFILE := $(MAKESUPPORT_OUTPUTDIR)/module-deps.gmk
 
 MODULE_INFOS := $(call FindAllModuleInfos, *)
 
-ifeq ($(TOOLCHAIN_TYPE),microsoft)
+ifeq ($(filter $(TOOLCHAIN_TYPE), microsoft xlc), $(TOOLCHAIN_TYPE))
```

Comment:
This surprises me: does xlc _really_ understand the `-nologo` option? It isn't documented at http://www-01.ibm.com/support/docview.wss?uid=swg27041854&aid=1.

Reply:
Not sure whether it understands it or does anything with it, it just seems to ignore it.
I think I will modify the code to have a different set of options for `xlc`

Reply:
According that documentation, `The -E option overrides the -P` so the latter can be omitted.
Also, elsewhere we use the 'else-if' feature of make (see https://www.gnu.org/software/make/manual/make.html#Conditional-Syntax) so it's safe to use here as well (resulting in a smaller change that's more easily extended if necessary):
```
ifeq ($(TOOLCHAIN_TYPE),microsoft)
  CPP_FLAGS_DEPS := -EP -nologo
else ifeq ($(TOOLCHAIN_TYPE),xlc)
  CPP_FLAGS_DEPS := -E
else
  CPP_FLAGS_DEPS := -P - <
endif
```